vaadin-login-flow-parent/vaadin-login-flow-integration-tests/src/test/java/com/vaadin/flow/component/login/test/OverlayClassNameIT.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2000-2022 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.vaadin.flow.component.login.test;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+import com.vaadin.flow.component.login.testbench.LoginOverlayElement;
+import com.vaadin.flow.testutil.TestPath;
+import com.vaadin.tests.AbstractComponentIT;
+
+@TestPath("vaadin-login/overlay-class-name")
+public class OverlayClassNameIT extends AbstractComponentIT {
+
+    private static final String LOGIN_OVERLAY_WRAPPER_TAG = "vaadin-login-overlay-wrapper";
+
+    @Before
+    public void init() {
+        open();
+    }
+
+    @Test
+    public void openDialog_overlayHasSameClassNames() {
+        findElement(By.id("open")).click();
+
+        waitForElementPresent(By.tagName(LOGIN_OVERLAY_WRAPPER_TAG));
+        LoginOverlayElement overlay = $(LoginOverlayElement.class).first();
+
+        WebElement overlayWrapper = $(LOGIN_OVERLAY_WRAPPER_TAG).first();
+
+        String overlayClassNames = overlay.getAttribute("class");
+        String wrapperClassNames = overlayWrapper.getAttribute("class");
+
+        Assert.assertEquals("custom", overlayClassNames);
+        Assert.assertEquals("custom", wrapperClassNames);
```

Comment:
This repeating code could be refactored into a method e.g. `assertClassEquals(overlay, classValue)`.
Basically, it's the same suggestion as the one for `Notification` PR here: https://github.com/vaadin/flow-components/pull/2931#discussion_r836326075